Dockerfile

```diff
@@ -24,38 +24,50 @@ RUN apk add --no-cache \
   tzdata \
   openssl \
   postgresql-dev \
-  libc6-compat
-
-RUN bundle install --jobs 4 --retry 5 && \
-  chmod -R a+r /usr/local/bundle
+  libc6-compat && \
+  bundle config --global without "${BUNDLE_WITHOUT}"  && \
+  bundle config --global path "${BUNDLE_PATH}" && \
+  bundle config --global deployment "${BUNDLE_DEPLOYMENT}" && \
+  bundle config --global retry 5 && \
+  bundle install && \
+  find /usr/local/bundle/ \
+    \( \
+      -name "*.c" -o \
+      -name "*.o" -o \
+      -name "*.a" -o \
+      -name "*.h" -o \
+      -name "Makefile" -o \
+      -name "*.md" \
+    \) -delete && \
+  chmod -R a+r "${BUNDLE_PATH}"
 
 # Final stage
 FROM base
 LABEL maintainer="keygen.sh <oss@keygen.sh>"
 
-WORKDIR /app
-COPY . /app
-
 RUN apk add --no-cache \
   bash \
   postgresql-client \
   tzdata \
-  libc6-compat
+  libc6-compat && \
+  adduser -h /app -g keygen -u 1000 -s /bin/bash -D keygen
```

Comment:
Let's move this into its own `RUN` command so that dependencies and permissions are clearly separate.

Reply:
Merging the RUN commands is the best practice recommended. I am using Hadolint, and it gives me a warning if I do that.  Should I proceed? 
https://github.com/hadolint/hadolint/wiki/DL3059#multiple-consecutive-run-instructions-consider-consolidation

Reply:
Ah, okay. No worries then. It was a nit anyways.